build/config/compiler/compiler.gni

```diff
@@ -43,7 +43,11 @@ declare_args() {
   c_standard = "gnu11"
 
   # C++ standard level (value for -std flag).
-  cpp_standard = "gnu++14"
+  if (current_os == "linux") {
+    cpp_standard = "gnu++17"
```

Comment:
This is likely going to cause people on Linux to write code that will then fail to compile for other platforms in CI...  Is there any way to avoid this?  Or condition it on perfetto being enabled, or limit to the perfetto bits or something?

Reply:
I could not find a good way to do this. CI would presumably catch this, however the issue I have is that perfetto is highly optimized and ends up having a lot of fancy things in a header, so since our tracing is global, when perfetto is enabled I globally start needing C++17 features.

At most I could imagine we want an app-by-app setup of this. Not sure how viable that is. I can try, for now trying to get all things to compile across all platforms, then I will take a look if I can place this in chip-tool only (I only have this gni and args.gni to set defaults).

Keeping this comment open.

Reply:
Right, if everyone can just move to c++17 that's obviously best.  If not, is there a way we can condition this on `matter_commandline_enable_perfetto_tracing`?  The layering might be tough here....

Reply:
In any case, handling this in a follow-up would be fine.

Reply:
@andy31415 Yes, that sounds great.  Thank you!

Reply:
Ended up keeping c++17 for android since all android builds should have perfetto (it is built-in in recent androids). The rest we will have to handle. I expect most people devleop on linux or mac first.